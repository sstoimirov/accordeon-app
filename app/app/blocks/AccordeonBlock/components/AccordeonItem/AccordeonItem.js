import * as React from "react";
export class AccordeonItem extends React.PureComponent {
    render() {
        const cls = "accordeon-text-holder";
        const activeCls = this.props.isOpened ? `${cls} active` : cls;
        return (React.createElement("div", { className: "accordeon-element-header", onClick: this.props.onClick },
            React.createElement("div", { className: "accordeon-title" }, this.props.isOpened ? this.props.title.opened : this.props.title.closed),
            React.createElement("div", { className: activeCls }, this.props.content)));
    }
}
//# sourceMappingURL=AccordeonItem.js.map