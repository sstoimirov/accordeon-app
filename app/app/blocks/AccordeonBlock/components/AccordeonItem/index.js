export * from "./AccordeonItem";
//# sourceMappingURL=index.js.map