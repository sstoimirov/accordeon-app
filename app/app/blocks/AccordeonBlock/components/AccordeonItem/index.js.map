{"version":3,"file":"index.js","sourceRoot":"","sources":["../../../../../src/blocks/AccordeonBlock/components/AccordeonItem/index.ts"],"names":[],"mappings":"AAAA,cAAc,iBAAiB,CAAC"}