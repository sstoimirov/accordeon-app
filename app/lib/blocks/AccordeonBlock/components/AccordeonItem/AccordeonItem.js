"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const React = __importStar(require("react"));
class AccordeonItem extends React.PureComponent {
    render() {
        const cls = "accordeon-text-holder";
        const activeCls = this.props.isOpened ? `${cls} active` : cls;
        return (React.createElement("div", { className: "accordeon-element-header", onClick: this.props.onClick },
            React.createElement("div", { className: "accordeon-title" }, this.props.isOpened ? this.props.title.opened : this.props.title.closed),
            React.createElement("div", { className: activeCls }, this.props.content)));
    }
}
exports.AccordeonItem = AccordeonItem;
//# sourceMappingURL=AccordeonItem.js.map