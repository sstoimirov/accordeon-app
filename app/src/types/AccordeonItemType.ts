export type AccordeonItemType = {
    id: string;
    title: {
        opened: string,
        closed: string
    };
    isOpened: boolean;
    content: string;
    onClick: (e:React.MouseEvent<HTMLDivElement>) => void;
}