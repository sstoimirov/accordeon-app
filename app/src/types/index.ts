export * from "./AccordeonType";
export * from "./AccordeonItemType";