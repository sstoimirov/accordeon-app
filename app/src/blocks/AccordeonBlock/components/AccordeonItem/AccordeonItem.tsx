import * as React from "react";
import { AccordeonItemType } from "../../../../types"

export class AccordeonItem extends React.PureComponent<AccordeonItemType, {}>{

    render() {
        const cls = "accordeon-text-holder"
        const activeCls = this.props.isOpened ? `${cls} active` : cls;
        return (
            <div className={"accordeon-element-header"} onClick={this.props.onClick}>
                <div className="accordeon-title">{this.props.isOpened ? this.props.title.opened : this.props.title.closed}</div>
                <div className={activeCls}>
                    {this.props.content}
                </div>
            </div >
        )
    }
}