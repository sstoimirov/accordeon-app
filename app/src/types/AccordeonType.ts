import { AccordeonItemType } from "./AccordeonItemType";

export type AccordeonType = {
    items: AccordeonItemType[],
    onClick: Function
}