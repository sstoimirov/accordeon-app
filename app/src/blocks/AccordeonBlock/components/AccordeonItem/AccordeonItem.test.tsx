import * as React from "react";
import { shallow, configure } from 'enzyme';
import { AccordeonItem } from "./AccordeonItem";
import { AccordeonItemType } from "../../../../types";
import Adapter from 'enzyme-adapter-react-16'

configure({ adapter: new Adapter() })

describe("AccordeonItem", () => {
    let props: AccordeonItemType;
    beforeEach(() => {
        props = {
            id: "122",
            title: { opened: "Opened", closed: "Closed" },
            isOpened: true,
            content: "Some content",
            onClick: jest.fn()
        }
    })
    it("Renders correclty with provided props and isOpened set to true", () => {
        const comp = shallow(<AccordeonItem {...props} onClick={jest.fn()} />)
        expect(comp).toMatchSnapshot();
    })

    it("Renders correclty with provided props and isOpened set to false", () => {
        const updatedProps = { ...props, isOpened: false }
        const comp = shallow(<AccordeonItem {...updatedProps} onClick={jest.fn()} />)
        expect(comp).toMatchSnapshot();
    })
})