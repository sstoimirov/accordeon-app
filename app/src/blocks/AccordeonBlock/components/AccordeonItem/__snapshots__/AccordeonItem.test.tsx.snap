// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccordeonItem Renders correclty with provided props and isOpened set to false 1`] = `ShallowWrapper {}`;

exports[`AccordeonItem Renders correclty with provided props and isOpened set to true 1`] = `ShallowWrapper {}`;
